discovery/kubernetes/service.go

```diff
@@ -233,7 +236,30 @@ func (s *Service) buildService(svc *apiv1.Service) *targetgroup.Group {
 		}
 
 		if svc.Spec.Type == apiv1.ServiceTypeLoadBalancer {
-			labelSet[serviceLoadBalancerIP] = lv(svc.Spec.LoadBalancerIP)
+			if len(svc.Status.LoadBalancer.Ingress) > 0 {
+				ips := make([]string, 0, len(svc.Status.LoadBalancer.Ingress))
+				hostnames := make([]string, 0, len(svc.Status.LoadBalancer.Ingress))
+				for _, ing := range svc.Status.LoadBalancer.Ingress {
+					if ing.IP != "" {
+						ips = append(ips, ing.IP)
+					}
+					if ing.Hostname != "" {
+						hostnames = append(hostnames, ing.Hostname)
+					}
+				}
+				if len(ips) > 0 {
+					labelSet[serviceLoadBalancerIPs] = lv(strings.Join(ips, ","))
+					if len(ips) == 1 {
+						labelSet[serviceLoadBalancerIP] = lv(ips[0])
+					}
+				}
+				if len(hostnames) > 0 {
+					labelSet[serviceLoadBalancerHostnames] = lv(strings.Join(hostnames, ","))
+				}
+			} else if svc.Spec.LoadBalancerIP != "" {
+				labelSet[serviceLoadBalancerIPs] = lv(svc.Spec.LoadBalancerIP)
+				labelSet[serviceLoadBalancerIP] = lv(svc.Spec.LoadBalancerIP)
+			}
```

Comment:
Am I wrong, or will this only add the `spec.LoadBalancer` value if the status is empty, making this effectively backwards incompatible?

I'm not sure if that's desirable, or is it?